### PR TITLE
CrowdStrike Falcon - RTR API

### PR DESCRIPTION
**Pull request by Itay4**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17035 https://github.com/demisto/etc/issues/18776

## Description
  - Implemented RTR API 
  - Added `email` argument to `resolve-detections` command

## Required version of Demisto
Any

## Does it break backward compatibility?
   - No

## Must have
- [x] Unit Tests
- [ ] Documentation
- [ ] Code Review

## Dependencies
None

## Technical writer review
- [ ] [YAML file](https://github.com/demisto/content/blob/18b912a7908e6f628c8d33e0f8bc4e9411651289/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.yml)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/18b912a7908e6f628c8d33e0f8bc4e9411651289/Integrations/CrowdStrikeFalcon/CHANGELOG.md)

### CODE REVIEW

**Comment by richardbluestone** [comment]
Reviewed and updated the docs

File: Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -844,30 +1090,361 @@ def lift_host_containment_command():
...
+    script_name = args.get('script_name')
+    raw = args.get('raw')
+    host_ids = argToList(args.get('host_ids'))
+
+    if script_name and raw:
+        raise ValueError('Only one of the arguments script_name or raw should be provided, not both.')
+    elif not script_name and not raw:
+        raise ValueError('One of the arguments script_name or raw must be provided, none given.')
+    elif script_name:
+        full_command = f'runscript -CloudFile={script_name}'
```
Comment (yaakovi) [inline Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py]: do we need to support names with spaces?

File: Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 2]
```diff
@@ -844,30 +1090,361 @@ def lift_host_containment_command():
...
+
+        return create_entry_object(contents=response, ec=entry_context, hr=human_readable)
+    else:
+        return 'No script found.'
+
+
+def delete_script_command():
+    script_id = demisto.args().get('script_id')
+
+    response = delete_script(script_id)
```
Comment (yaakovi) [inline Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py]: can this fail without exception? if so, hr should be set accordingly

File: Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 3]
```diff
@@ -844,30 +1090,361 @@ def lift_host_containment_command():
...
+
+
+def get_script_command():
+    script_id = argToList(demisto.args().get('script_id'))
+
+    response = get_script(script_id)
+
+    resources: list = response.get('resources', [])
+    if resources and isinstance(resources, list):
+        resource = resources[0]
```
Comment (yaakovi) [inline Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py]: what about the rest of the list? if it's only one, add a comment to specify that

File: Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 4]
```diff
@@ -844,30 +1090,361 @@ def lift_host_containment_command():
...
+    permission_type = args.get('permission_type', 'private')
+    content = args.get('content')
+    entry_id = args.get('entry_id')
+
+    if content and entry_id:
+        raise ValueError('Only one of the arguments entry_id or content should be provided, not both.')
+    elif not content and not entry_id:
+        raise ValueError('One of the arguments entry_id or content must be provided, none given.')
+
+    response = upload_script(name, permission_type, content, entry_id)
```
Comment (yaakovi) [inline Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py]: can this fail without exception? if so, hr should be set accordingly

File: Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py (hunk 1 above)
Comment (Itay4) [reply Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py]: what do you mean?

File: Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 5]
```diff
@@ -844,30 +1090,361 @@ def lift_host_containment_command():
...
+            'Stdout': resource.get('stdout'),
+            'Stderr': resource.get('stderr'),
+            'BaseCommand': resource.get('base_command'),
+            'Command': full_command
+        })
+
+    human_readable = tableToMarkdown(f'Command {full_command} results', output)
+    entry_context = {
+        'CrowdStrike': {
+            'Command': output
```
[the remark this answers is not in the document]
Comment (Itay4) [reply Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py]: there isn't really a unique field here

**Pull request merged**